src/badge/minter.cairo

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin, SignatureBuiltin
+from starkware.cairo.common.uint256 import Uint256
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.math import assert_not_zero
+from starkware.cairo.common.alloc import alloc
+
+// Project dependencies
+from openzeppelin.access.ownable.library import Ownable
+from openzeppelin.introspection.erc165.library import ERC165
+from openzeppelin.upgrades.library import Proxy
+from bal7hazar.token.erc1155.library import ERC1155
+
+// Local dependencies
+from src.badge.library import CarbonableBadge
+from src.badge.Ibadge import IBadgeContract
+
+//
+// Storage
+//
+
+@storage_var
+func signer_public_key() -> (signer_public_key: felt) {
+}
+
+@storage_var
+func badge_contract_address() -> (badge_contract_address: felt) {
+}
+
+//
+// Constructor
+//
+
+@constructor

```

Comment:
No constructor decorator for upgradable contracts, un peu de doc au besoin :
- https://docs.openzeppelin.com/contracts-cairo/0.5.0/proxies
- https://ctrlc03.github.io/